v1.0: Primary key inference (#2104)
meilisearch/documentation: reference/errors/error_codes.md

The requested index already has a primary key that cannot be changed.

## `index_primary_key_multiple_candidates_found`

> [Primary key inference](/learn/core_concepts/primary_key.md#meilisearch-guesses-your-primary-key) failed as the index contains multiple fields ending with `id`. Use the [update index endpoint](/reference/api/indexes.md#update-an-index) to manually set a primary key.

Reviewer: ```suggestion
[Primary key inference](/learn/core_concepts/primary_key.md#meilisearch-guesses-your-primary-key) failed because the received documents contain multiple fields ending with `id`. Use the [update index endpoint](/reference/api/indexes.md#update-an-index) to manually set a primary key.
```

> [Primary key inference](/learn/core_concepts/primary_key.md#meilisearch-guesses-your-primary-key) failed as the index contains multiple fields ending with `id`. Use the [update index endpoint](/reference/api/indexes.md#update-an-index) to manually set a primary key or add `id` to one of your fields so it can be used as the primary key during inference. We recommend manually setting the primary key.

Reviewer 2: AFAIK adding `id` won't work in this case because we already have multiple candidates, and `id` is not given any priority.

As it probably doesn't make sense for users to remove fields from their documents, I'd say that manually specifying the field representing the primary key is their only recourse when there are multiple candidates.

Adding `id` to one of the fields **does** work in the `index_primary_key_no_candidate_found` case, though.

The field you are trying to modify is immutable.

## `index_primary_key_no_candidate_found`

> [Primary key inference](/learn/core_concepts/primary_key.md#meilisearch-guesses-your-primary-key) failed as the index does not contain any fields ending with `id`. [Manually designate the primary key](/learn/core_concepts/primary_key.md#setting-the-primary-key) or add `id` to one of your fields so it can be used as the primary key during inference. We recommend manually setting the primary key.

Reviewer: I don't think we should make this recommendation, as their dataset might not contain a field suitable to be used as a primary key (i.e. present in every document and containing unique values).

```suggestion
[Primary key inference](/learn/core_concepts/primary_key.md#meilisearch-guesses-your-primary-key) failed as the received documents do not contain any fields ending with `id`. [Manually designate the primary key](/learn/core_concepts/primary_key.md#setting-the-primary-key), or add some field ending with `id` to your documents.
```
